azurelinuxagent/ga/remoteaccess.py

```diff
@@ -102,60 +102,78 @@ def run(self):
+                        # user explicitly removed
+                        logger.info("User {0} removed from remote_access".format(user))
+                        self.remove_user(user)
+                except RemoteAccessError as rae:
+                    self.err_message = self.err_message + "Error removing user {0}. Exception: {1}"\
+                        .format(user, ustr(rae))
+        else:
+            # All users removed, remove any remaining JIT accounts.
+            for user in existing_jit_users:
+                try:
+                    logger.info("User {0} removed from remote_access. remote_access empty".format(user))
+                    self.remove_user(user)
+                except RemoteAccessError as rae:
+                    self.err_message = self.err_message + "Error removing user {0}. Exception: {1}"\
+                        .format(user, ustr(rae))
 
     def validate_jit_user(self, comment):
         return comment == REMOTE_ACCESS_ACCOUNT_COMMENT
 
     def add_user(self, username, encrypted_password, account_expiration):
         try:
             expiration_date = (account_expiration + timedelta(days=1)).strftime(DATE_FORMAT)
-            logger.verbose("Adding user {0} with expiration date {1}"
-                           .format(username, expiration_date))
+            logger.verbose("Adding user {0} with expiration date {1}".format(username, expiration_date))
             self.os_util.useradd(username, expiration_date, REMOTE_ACCESS_ACCOUNT_COMMENT)
-        except OSError as oe:
-            logger.error("Error adding user {0}. {1}"
-                         .format(username, oe.strerror))
-            return
         except Exception as e:
-            logger.error("Error adding user {0}. {1}".format(username, ustr(e)))
-            return
+            raise RemoteAccessError("Error adding user {0}. {1}".format(username, ustr(e)))
         try:
             prv_key = os.path.join(conf.get_lib_dir(), TRANSPORT_PRIVATE_CERT)
             pwd = self.cryptUtil.decrypt_secret(encrypted_password, prv_key)
             self.os_util.chpasswd(username, pwd, conf.get_password_cryptid(), conf.get_password_crypt_salt_len())
             self.os_util.conf_sudoer(username)
-            logger.info("User '{0}' added successfully with expiration in {1}"
-                        .format(username, expiration_date))
-            return
-        except OSError as oe:
-            self.handle_failed_create(username, oe.strerror)
+            logger.info("User '{0}' added successfully with expiration in {1}".format(username, expiration_date))
         except Exception as e:
-            self.handle_failed_create(username, ustr(e))
+            try:
+                self.handle_failed_create(username)
+            except RemoteAccessError as rae:
+                raise RemoteAccessError("Error adding user {0} and error cleaning up {1}".format(username, ustr(e)), rae)
```

Comment:
I might suggest simplifying this to a single raise, and setting flags based on what was successful